NetSDK/Services/Client/Classes/SplitClient.cs

```diff
@@ -55,37 +58,72 @@ private void RecordStats(Key key, string feature, long start, string treatment,
 
             if (treatmentLog != null)
             {
-                treatmentLog.Log(key.matchingKey, feature, treatment, start, key.bucketingKey);
+                treatmentLog.Log(key.matchingKey, feature, treatment, start, changeNumber, label, key.bucketingKey);
             }
         }
 
         private string GetTreatmentForFeature(Key key, string feature, Dictionary<string, object> attributes)
         {
+            long start = CurrentTimeHelper.CurrentTimeMillis();
+            var clock = new Stopwatch();
+            clock.Start();
+
             try
             {
                 var split = splitCache.GetSplit(feature);
 
                 if (split == null)
                 {
+                    //if split definition was not found, impression label = "rules not found"
+                    RecordStats(key, feature, null, LabelSplitNotFound, start, Control, SdkGetTreatment, clock);
+
                     Log.Warn(String.Format("Unknown or invalid feature: {0}", feature));
                     return Control;
                 }
-
-                long start = CurrentTimeHelper.CurrentTimeMillis();
-                var clock = new Stopwatch();
-                clock.Start();
                 
-                var treatment = engine.GetTreatment(key, split, attributes);
+                var treatment = GetTreatment(key, split, attributes, start, clock);
                 
-                RecordStats(key, feature, start, treatment, SdkGetTreatment, clock);
-
                 return treatment;
             }
             catch (Exception e)
             {
+                //if there was an exception, impression label = "exception"
+                RecordStats(key, feature, null, LabelException, start, Control, SdkGetTreatment, clock);
+
                 Log.Error(String.Format("Exception caught getting treatment for feature: {0}", feature), e);
                 return Control;
             }
         }
+
+        private string GetTreatment(Key key, ParsedSplit split, Dictionary<string, object> attributes, long start, Stopwatch clock)
+        {
+            if (!split.killed)
+            {
+                foreach (ConditionWithLogic condition in split.conditions)
+                {
+                    var combiningMatcher = condition.matcher;
+                    if (combiningMatcher.Match(key.matchingKey, attributes))
+                    {
+                        var treatment = splitter.GetTreatment(key.bucketingKey, split.seed, condition.partitions);
```

Comment:
Yes, we had. I think it was implemented one of the last days on the previous stage.